packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -117,6 +99,61 @@ export const getLetterEditingOptions = <TDate>(
   }
 };
 
+export const cleanDigitSectionValue = <TDate>(
+  utils: MuiPickersAdapter<TDate>,
+  value: number,
+  sectionType: FieldSectionType,
+  format: string,
+  hasLeadingZeros: boolean,
+  sectionBoundaries: FieldSectionValueBoundaries<TDate, any>,
+) => {
+  const hasLetterPrefix = () => {
+    const startOfYear = utils.startOfYear(utils.date()!);
+    const startOfYearStr = utils.formatByString(startOfYear, format);
+
+    return Number.isNaN(Number(startOfYearStr));
```

Comment:
I'm not sure about this function's name correctness. 🤔 
It does not seem to check for the prefix in any way, and both `1st` as well as `st1` would return `true` in this case.
Maybe just rename the function to `hasLetters`?